packages/peer-exchange/src/waku_peer_exchange_discovery.ts

```diff
@@ -46,21 +62,25 @@ export class PeerExchangeDiscovery
   private readonly peerExchange: WakuPeerExchange;
   private readonly options: Options;
   private isStarted: boolean;
-  private intervals: Map<PeerId, NodeJS.Timeout> = new Map();
+  private intervals: Map<string, NodeJS.Timeout> = new Map();
+  private queryAttempts: Map<string, number> = new Map();
 
   private readonly eventHandler = async (
     event: CustomEvent<PeerProtocolsChangeData>
   ): Promise<void> => {
     const { protocols, peerId } = event.detail;
-    if (!protocols.includes(PeerExchangeCodec) || this.intervals.get(peerId))
+    if (
+      !protocols.includes(PeerExchangeCodec) ||
+      this.intervals.get(peerId.toString())
+    )
       return;
 
-    await this.query(peerId);
-    const interval = setInterval(async () => {
-      await this.query(peerId);
-    }, PEER_EXCHANGE_QUERY_INTERVAL);
-
-    this.intervals.set(peerId, interval);
+    this.startRecurringQueries(
+      peerId,
+      this.options.initialQueryTimeout,
+      this.options.attemptsBeforeBackOff,
+      this.options.queryInterval
```

Comment:
`startRecurringQueries` is a method of this class, and hence has access to `this`, so you don't need to pass the `this.options` values.